### PR TITLE
Disable password encode in PDS. Remove {noop} from profiles. #673

### LINKED ISSUE #673: Add `{noop}` to PDS Secret if missing
**Problem:**

The product delegation server (PDS) does not save user passwords in a database. Instead the secrets are injected at runtime (e.g. container start). In addition, the PDS has only two users `admin` and `techuser`.

SpringBoot uses strings to indicate how the password is checked/stored. The string to indicate no hashing is used, is `{noop}`. The string needs to be added to every PDS password. In case, one forgets to add the `{noop}` the password will not work as the default SpringBoot implementation assumes the password is stored as Bcrypt hash.

**Solution:**

In case, the PDS password is not prefixed by `{noop}` add the prefix to it.

### PR DESCRIPTION
**Pull request by zigfridus**
Disable password encode in PDS. 
Remove {noop} from profiles. 
Closes #673

### CODE REVIEW

**Comment by zigfridus** [comment]
@Jeeppler 
Please review this PR.

**Comment by Jeeppler** [comment]
@de-jcup I assigned you as reviewer.

File: sechub-pds/src/main/resources/application-pds_integrationtest.yml [hunk 1]
```diff
@@ -29,10 +29,10 @@ sechub:
                           # time
    techuser:
        userid: pds-inttest-techuser
-       apitoken: "{noop}pds-inttest-apitoken"
```
Comment (de-jcup) [inline sechub-pds/src/main/resources/application-pds_integrationtest.yml:32]: This is fine, so we test the new "add noop when missing" mechanism automatically when integration tests are executed by GitHub actions.

File: sechub-pds/src/main/java/com/mercedesbenz/sechub/pds/security/PDSSecurityConfiguration.java [hunk 2]
```diff
@@ -53,24 +53,27 @@ public class PDSSecurityConfiguration extends AbstractAllowPDSAPISecurityConfigu
...
-                .roles(PDSRoles.USER.getRole())
-                .build();
+                User.builder()
+                        .username(techUserId)
+                        .password(disablePasswordEncode(techUserApiToken))
+                        .roles(PDSRoles.USER.getRole())
+                        .build();
         /* remove field after start */
-        techUserApiToken=null;
+        techUserApiToken = null;
```
Comment (de-jcup) [inline sechub-pds/src/main/java/com/mercedesbenz/sechub/pds/security/PDSSecurityConfiguration.java:62]: Just a reminder: Please do always execute `./gradlew spotlessApply` to automatically format source code before you push to Git.

**Review by de-jcup — CHANGES_REQUESTED** [review]
There are some changes necessary. Please read my comments.

**Comment by zigfridus** [comment]
Hello @de-jcup 
I changed my PR according to your comments. 
Please have a look.

File: sechub-pds/src/test/java/com/mercedesbenz/sechub/pds/security/PDSPasswordTransformerTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    @Test
+    public void skip_password_with_bcrypt() {
+        String originPassword = "{bcrypt}my-pwd";
+        String transformedPassword = passwordTransformer.transformPassword(originPassword);
+        assertEquals("{bcrypt}my-pwd", transformedPassword);
+    }
+
+}
```
Comment (de-jcup) [inline sechub-pds/src/test/java/com/mercedesbenz/sechub/pds/security/PDSPasswordTransformerTest.java:39]: After adding the null argument check implementaiton, please add a junit5 test method like this:

```suggestion
  @Test
  void null_argument() {
     /* execute + test */
     assertThrows(IllegalArgumentException.class, () -> {
         transformerToTest.transformPassword(null);
     });
  }
}
```

**Comment by zigfridus** [comment]
@de-jcup
I made changes in my code according to your comments.
Please have a look.

**Review by de-jcup — APPROVED** [review]
Good job :rocket: 
Thank you for the contribution.

**Pull request merged**